controller_manager/test/test_resource_manager.cpp

```diff
@@ -221,3 +224,151 @@ TEST_F(TestResourceManager, initialization_with_wrong_urdf) {
     SUCCEED() << e.what();
   }
 }
+
+TEST_F(TestResourceManager, initialization_with_urdf_unclaimed) {
+  auto urdf = urdf_head_ + test_hardware_resource_system_ + urdf_tail_;
+  // we validate the results manually
+  controller_manager::ResourceManager rm(urdf);
+
+  auto command_interface_keys = rm.command_interface_keys();
+  for (const auto & key : command_interface_keys) {
+    EXPECT_FALSE(rm.command_interface_is_claimed(key));
+  }
+  // state interfaces don't have to be locked, hence any arbitrary key
+  // should return false.
+  auto state_interface_keys = rm.state_interface_keys();
+  for (const auto & key : state_interface_keys) {
+    EXPECT_FALSE(rm.command_interface_is_claimed(key));
+  }
+}
+
+TEST_F(TestResourceManager, resource_claiming) {
+  auto urdf = urdf_head_ + test_hardware_resource_system_ + urdf_tail_;
+  controller_manager::ResourceManager rm(urdf);
+
+  const auto key = "joint1/position";
+  EXPECT_FALSE(rm.command_interface_is_claimed(key));
+
+  {
+    auto position_command_interface = rm.claim_command_interface(key);
+    EXPECT_TRUE(rm.command_interface_is_claimed(key));
+    try {
+      auto sensor1_again = rm.claim_command_interface(key);
+      FAIL();
+    } catch (const std::runtime_error &) {
+      SUCCEED();
+    }
+  }
+  EXPECT_FALSE(rm.command_interface_is_claimed(key));
+
+  // command interfaces can only be claimed once
+  for (const auto & key :
+    {"joint1/position", "joint1/position", "joint1/position", "joint2/velocity",
+      "joint3/velocity"})
+  {
+    {
+      auto interface = rm.claim_command_interface(key);
+      EXPECT_TRUE(rm.command_interface_is_claimed(key));
+      try {
+        auto interface_again = rm.claim_command_interface(key);
+        FAIL();
```

Comment:
we can replace these with EXPECT_THROW in a follow-up PR

Reply:
done